src/agent/src/rpc.rs

```diff
@@ -124,6 +125,34 @@ pub fn verify_cid(id: &str) -> Result<()> {
     }
 }
 
+// merge oci process
+fn merge_process(ori: &mut oci::Process, image: &oci::Process) {
+    if ori.args.is_empty() && !image.args.is_empty() {
+        ori.args.append(&mut image.args.clone());
+    }
+    if ori.cwd.is_empty() {
+        ori.cwd = image.cwd.clone();
+    }
+    ori.env.append(&mut image.env.clone());
+}
+
+// merge oci spec
+fn merge_oci(image_bundle: &str, ori: &mut oci::Spec, image: &oci::Spec) {
+    if let Some(root) = ori.root.as_mut() {
+        if let Some(ir) = image.root.as_ref() {
+            let mut rp = PathBuf::from(CONTAINER_BASE);
+            rp.push(image_bundle);
+            rp.push(ir.path.clone());
+            root.path = String::from(rp.to_str().unwrap());
+        }
+    }
+    if let Some(proc) = ori.process.as_mut() {
+        if let Some(image_proc) = image.process.as_ref() {
+            merge_process(proc, image_proc);
+        }
+    }
+}
+
```

Comment:
I find this piece of code a little confusing, mostly because of naming. 
Here is a suggested improvement:

```rust
// Partially merge an OCI process specification into another one.
fn merge_oci_process(target: &mut oci::Process, source: &oci::Process) {
    if target.args.is_empty() && !source.args.is_empty() {
        target.args.append(&mut source.args.clone());
    }

    if target.cwd.is_empty() && !source.cwd.is_empty(){
        target.cwd = String::from(&source.cwd);
    }

    target.env.append(&mut source.env.clone());
}

// When being passed an image name through a container annotation, merge its
// corresponding bundle OCI specification into the passed container creation one.
async fn merge_bundle_oci(&self, container_oci: &oci::Spec) -> Result<()> {
        if let Some(image_name) = container_oci.annotations.get(&ANNO_K8S_IMAGE_NAME.to_string()) {
                if let Some(container_id) = self.sandbox.clone().lock().await.images.get(image_name) {
                     let image_oci_config_path = PathBuf::from(CONTAINER_BASE);
                     image_oci_config_path.push(&container_id);
                     image_oci_config_path.push(&String::from("config.json"));
                     debug!(sl!(), "Image bundle config path: {}", image_oci_config_path);
                
                     let image_oci = oci::Spec::load(&image_oci_config_path).context("load image bundle")?;
 
                     if let Some(container_root) = container_oci.root.as_mut() {
                          if let Some(image_root) = image_oci.root.as_ref() {
                              let mut root_path = PathBuf::from(CONTAINER_BASE);
                              root_path.push(container_id);
                              root_path.push(image_root.path.clone());
                              container_root.path = String::from(root_path.to_str().unwrap());
                          }
                     }

                     if let Some(container_process) = container_oci.process.as_mut() {
                          if let Some(image_process) = image_oci.process.as_ref() {
                              merge_oci_process(container_process, image_process);
                          }
                     }
                }
            }

        Ok(())
}
```

Reply:
I added that implementation to your branch @wllenyj .

Please let me know if it's ok with you.

Reply:
LGTM